lazycoder1412/GenerateParentheses.cpp

```diff
@@ -0,0 +1,18 @@
+class Solution {
+public:
+    vector<string> ans;
+    void backtrack(string temp, int open , int close, int n){
```

Comment:
`string& temp` and at the end of the function `temp.pop_back()`. Using reference makes the code faster by not copying every time.

Reply:
Actually, that will throw an error because the variable being passed is `temp+'('` The correct way to do this would be to create a variable `next_string = current_string + '('` and then pass it as a `const string &temp`

Reply:
Yeah true. Didn't see that.